File: HydraScan/HydraScan/Models/ClientProfile.swift
import Foundation

enum BodyRegion: String, Codable, CaseIterable, Hashable, Identifiable {
    case rightShoulder = "right_shoulder"
    case leftShoulder = "left_shoulder"
    case rightHip = "right_hip"
    case leftHip = "left_hip"
    case lowerBack = "lower_back"
    case upperBack = "upper_back"
    case rightKnee = "right_knee"
    case leftKnee = "left_knee"
    case neck
    case rightCalf = "right_calf"
    case leftCalf = "left_calf"
    case rightArm = "right_arm"
    case leftArm = "left_arm"
    case rightFoot = "right_foot"
    case leftFoot = "left_foot"

    var id: String { rawValue }

    var displayLabel: String {
        switch self {
        case .rightShoulder: return "Right Shoulder"
        case .leftShoulder: return "Left Shoulder"
        case .rightHip: return "Right Hip"
        case .leftHip: return "Left Hip"
        case .lowerBack: return "Lower Back"
        case .upperBack: return "Upper Back"
        case .rightKnee: return "Right Knee"
        case .leftKnee: return "Left Knee"
        case .neck: return "Neck"
        case .rightCalf: return "Right Calf"
        case .leftCalf: return "Left Calf"
        case .rightArm: return "Right Arm"
        case .leftArm: return "Left Arm"
        case .rightFoot: return "Right Foot"
        case .leftFoot: return "Left Foot"
        }
    }
}

enum RecoveryGoal: String, Codable, CaseIterable, Hashable, Identifiable {
    case mobility
    case warmUp = "warm_up"
    case recovery
    case relaxation
    case performancePrep = "performance_prep"

    var id: String { rawValue }

    var displayLabel: String {
        switch self {
        case .mobility: return "Improve Mobility"
        case .warmUp: return "Warm Up for Movement"
        case .recovery: return "Support Recovery"
        case .relaxation: return "Relax and Reset"
        case .performancePrep: return "Prepare for Performance"
        }
    }
}

enum RecoverySignalType: String, Codable, CaseIterable, Hashable, Identifiable {
    case stiffness
    case soreness
    case tightness
    case restriction
    case guarding

    var id: String { rawValue }
}

enum ActivityTrigger: String, Codable, CaseIterable, Hashable, Identifiable {
    case morning
    case afterRunning = "after_running"
    case afterLifting = "after_lifting"
    case postTravel = "post_travel"
    case postTraining = "post_training"
    case evening
    case general

    var id: String { rawValue }
}

struct RecoverySignalValue: Codable, Hashable {
    var type: RecoverySignalType
    var severity: Int
    var trigger: String
    var notes: String?
}

struct RecoverySignal: Identifiable, Codable, Hashable {
    var id: String { region.rawValue }
    var region: BodyRegion
    var type: RecoverySignalType
    var severity: Int
    var trigger: String
    var notes: String?

    var value: RecoverySignalValue {
        RecoverySignalValue(type: type, severity: severity, trigger: trigger, notes: notes)
    }
}

struct ClientProfile: Identifiable, Codable, Hashable {
    var id: UUID
    var userID: UUID
    var clinicID: UUID?
    var primaryRegions: [BodyRegion]
    var recoverySignalsByRegion: [String: RecoverySignalValue]
    var goals: [RecoveryGoal]
    var activityContext: String?
    var sensitivities: [String]
    var notes: String?
    var wearableHRV: Double?
    var wearableStrain: Double?
    var wearableSleepScore: Double?
    var wearableLastSync: Date?
    var createdAt: Date
    var updatedAt: Date

    var recoverySignals: [RecoverySignal] {
        primaryRegions.compactMap { region in
            guard let value = recoverySignalsByRegion[region.rawValue] else {
                return nil
            }

            return RecoverySignal(
                region: region,
                type: value.type,
                severity: value.severity,
                trigger: value.trigger,
                notes: value.notes
            )
        }
    }

    static let empty = ClientProfile(
        id: UUID(),
        userID: UUID(),
        clinicID: nil,
        primaryRegions: [],
        recoverySignalsByRegion: [:],
        goals: [],
        activityContext: nil,
        sensitivities: [],
        notes: nil,
        wearableHRV: nil,
        wearableStrain: nil,
        wearableSleepScore: nil,
        wearableLastSync: nil,
        createdAt: Date(),
        updatedAt: Date()
    )
}

File: shared/src/types/client-profile.ts
export type BodyRegion =
  | "right_shoulder"
  | "left_shoulder"
  | "right_hip"
  | "left_hip"
  | "lower_back"
  | "upper_back"
  | "right_knee"
  | "left_knee"
  | "neck"
  | "right_calf"
  | "left_calf"
  | "right_arm"
  | "left_arm"
  | "right_foot"
  | "left_foot";

export type RecoveryGoal =
  | "mobility"
  | "warm_up"
  | "recovery"
  | "relaxation"
  | "performance_prep";

export type RecoverySignalType =
  | "stiffness"
  | "soreness"
  | "tightness"
  | "restriction"
  | "guarding";

export type ActivityTrigger =
  | "morning"
  | "after_running"
  | "after_lifting"
  | "post_travel"
  | "post_training"
  | "evening"
  | "general";

export interface RecoverySignal {
  region: BodyRegion;
  type: RecoverySignalType;
  severity: number;
  trigger: ActivityTrigger | string;
  notes?: string;
}

export interface RecoverySignalValue {
  type: RecoverySignalType;
  severity: number;
  trigger: ActivityTrigger | string;
  notes?: string;
}

export type RecoverySignalsByRegion = Partial<Record<BodyRegion, RecoverySignalValue>>;

export interface ClientProfileRecord {
  id: string;
  user_id: string;
  clinic_id: string;
  primary_regions: BodyRegion[];
  recovery_signals: RecoverySignalsByRegion;
  goals: RecoveryGoal[];
  activity_context: string | null;
  sensitivities: string[];
  notes: string | null;
  wearable_hrv: number | null;
  wearable_strain: number | null;
  wearable_sleep_score: number | null;
  wearable_last_sync: string | null;
  created_at: string;
  updated_at: string;
}
